Collection not opened - custom study

I can't use custom study. Every time I open the deck and click on a card (to see the answer) I get the following message:

"Collection not opened - An error occurred, while accessing your collection. It may have been caused by a harmless bug, or there might be a problem with it.

Try running check database, or restoring from backup by pressing "options" button. If you continue to see this message, please file a bug report."

I did the abovementioned steps, but it didn't help. What else can I try to be able to enter custom study again?

My bug report is this:

AnkiDroid Version = 2.9.1

Android Version = 8.0.0

ACRA UUID = 5d6e2fc5-d664-46b2-b69f-cbb0be3fa9e8


###### Reproduction Steps

1. Running check data base
2. Restoring all decks from backup
3. Deleting custom study deck and tried again


###### Expected Result

I hoped to be able to access custom study.

###### Actual Result

I receive the following message: 
"Collection not opened - An error occurred, while accessing your collection. It may have been caused by a harmless bug, or there might be a problem with it..."

###### Debug info

AnkiDroid Version = 2.9.1

Android Version = 8.0.0

ACRA UUID = 5d6e2fc5-d664-46b2-b69f-cbb0be3fa9e8


###### Research
*Enter an [ x ] character to confirm the points below:*

x

[  ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[  ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[  ] I have searched for similar existing issues here and on the user forum

If it says 2.10alpha30 then you are on the latest alpha, great

I'm sorry it is still happening. Using the UUID you posted I was able to find the bug report in our crash database:

```
java.lang.RuntimeException: org.json.JSONException: No value for resched
at com.ichi2.libanki.Sched._lapseConf(Sched.java:14)
at com.ichi2.libanki.Sched._lrnConf(Sched.java:2)
at com.ichi2.libanki.Sched.answerButtons(Sched.java:3)
at com.ichi2.anki.AbstractFlashcardViewer.showEaseButtons(AbstractFlashcardViewer.java:2)
at com.ichi2.anki.AbstractFlashcardViewer.displayCardAnswer(AbstractFlashcardViewer.java:20)
at com.ichi2.anki.AbstractFlashcardViewer$4.onClick(AbstractFlashcardViewer.java:5)
at android.view.View.performClick(View.java:6897)
at android.view.View$PerformClick.run(View.java:26104)
at android.os.Handler.handleCallback(Handler.java:789)
at android.os.Handler.dispatchMessage(Handler.java:98)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6944)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
Caused by: org.json.JSONException: No value for resched
at org.json.JSONObject.get(JSONObject.java:392)
at org.json.JSONObject.getBoolean(JSONObject.java:413)
at com.ichi2.libanki.Sched._lapseConf(Sched.java:13)
... 14 more
```

it looks like somehow your collection has empty values for things that should not be empty.

Is it possible for you to rebuild that custom study deck? The value that is unexpectedly empty seems like it has to do with the learning intervals, which I think are stored in the deck or the deck options group

--- a/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java
@@ -1773,14 +1773,19 @@ public abstract class AbstractFlashcardViewer extends NavigationDrawerActivity {
             mShowNextReviewTime = getCol().getConf().getBoolean("estTimes");
             mShowRemainingCardCount = getCol().getConf().getBoolean("dueCounts");
 
-            // Get the review options for this group
-            JSONObject revOptions = getCol().getDecks().confForDid(getCol().getDecks().selected()).getJSONObject("rev");
+            // Dynamic don't have review options; attempt to get deck-specific auto-advance options
+            // but be prepared to go with all default if it's a dynamic deck
+            JSONObject revOptions = new JSONObject();
+            if (!getCol().getDecks().isDyn(getCol().getDecks().current().getLong("id"))) {
+                revOptions = getCol().getDecks().confForDid(getCol().getDecks().current().getLong("id")).getJSONObject("rev");
+            }
 
             mOptUseGeneralTimerSettings = revOptions.optBoolean("useGeneralTimeoutSettings", true);
             mOptUseTimer = revOptions.optBoolean("timeoutAnswer", false);
             mOptWaitAnswerSecond = revOptions.optInt("timeoutAnswerSeconds", 20);
             mOptWaitQuestionSecond = revOptions.optInt("timeoutQuestionSeconds", 60);
         } catch (JSONException e) {
+            Timber.e(e, "Unable to restoreCollectionPreferences");
             throw new RuntimeException(e);
         } catch (NullPointerException npe) {
             // NPE on collection only happens if the Collection is broken, follow AnkiActivity example

